akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberEntities.scala

```diff
@@ -27,12 +27,15 @@ import akka.persistence.SnapshotSelectionCriteria
  * INTERNAL API
  */
 @InternalApi
-private[akka] final class EventSourcedRememberEntitiesStoreProvider(typeName: String, settings: ClusterShardingSettings)
-    extends RememberEntitiesShardStoreProvider {
+private[akka] final class EventSourcedRememberEntitiesProvider(typeName: String, settings: ClusterShardingSettings)
+    extends RememberEntitiesProvider {
 
   override def shardStoreProps(shardId: ShardId): Props =
     EventSourcedRememberEntitiesStore.props(typeName, shardId, settings)
 
+  // FIXME persistent state store deprecated but we are adding a remember entities store that is not deprecated
+  // We need a new impl for this to allow ddata + persistent remember entities
```

Comment:
One challenge here is that if the event sourced actor fails (persist failure) it must still continue retrying so that AddShard or GetShards are not stuck forever.

Reply:
Hmm, so perhaps better to move the retrying to the coordinator after all.

Would be nice to align how failure works with the remember entity store for the shard, so maybe not let-it-crash there either.

Reply:
The reason I pushed it into the store is that what makes sense with retrying, timeouts etc, kind of depends a bit on the store itself.

Reply:
Otherwise the retry logic would be in a parent supervisor of the event sourced store. It can keep current command around and retry.

Reply:
I updated the protocol with a fail + retrying on timeout because I think that is more clear.